enclave-tls/src/tls_wrappers/openssl/negotiate.c

```diff
@@ -43,7 +46,14 @@ tls_wrapper_err_t openssl_internal_negotiate(tls_wrapper_ctx_t *ctx, unsigned lo
 		return -TLS_WRAPPER_ERR_NO_MEM;
 
 	X509_STORE *cert_store = SSL_CTX_get_cert_store(ssl_ctx->sctx);
-	X509_STORE_set_ex_data(cert_store, 0, ctx);
+	int etls_ex_data_idx = X509_STORE_get_ex_new_index(0, "etls_ex_data", NULL, NULL, NULL);
+	X509_STORE_set_ex_data(cert_store, etls_ex_data_idx, ctx);
+
```

Comment:
move `int *etls_ex_data = calloc(1, sizeof(int));` to here, and change to `int *ex_data = calloc(1, sizeof(*etls_ex_data));`